packages/ui/core-components/src/lib/unsorted/viz/map/EvidenceMap.js

```diff
@@ -431,4 +499,84 @@ export class EvidenceMap {
 
 		return data;
 	}
+
+	handleLegendValues(colorPalette, values, legendType) {
+		//determine legend style
+		if (legendType === 'category') {
+			let uniqueValues = new Set(values);
+			values = [...uniqueValues];
+			let i = 0;
+
+			while (colorPalette.length < values.length) {
+				if (!backupColors[i]) {
+					throw new Error('No more backup colors available.');
+				}
```

Comment:
I think its better to wrap around (i.e. `backupColors[i % backupColors.length]`) and reuse colors rather than throw an error

Reply:
**Handling End of Color Options:**

@hughess, how do you feel about managing categorical values that exceed available color options, including both the color palette and backup colors?

@zachstence suggested avoiding throwing an error, instead opting to indicate a warning to the user.

We are considering using chroma.random() to generate new distinct colors. This approach would ensure that colors are always assigned to any number of categories, effectively handling dynamic categorical lists when we reach the end of the color palette and backup colors.

**_Caveat:_** The randomly generated colors would be independent on each refresh and across different browsers. For instance, the colors generated on your browser may differ from those generated on mine.

Reply:
I would have it start at the beginning of whatever the first color palette is. So if they pass a custom palette, it should go back to the first color in that palette and repeat. 

If they don't pass a custom palette, it should go back to the first built-in color and go from there.

This is the same way the charts operate at the moment. In the future, if it becomes an issue we can just add a bunch of colors to the built-in palette to accommodate longer lists